contracts/Strategy.sol

```diff
@@ -11,36 +11,138 @@ import {
     Address
 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 
-import "./libraries/MakerDaiDelegateLib.sol";
+import "./libraries/Boring/BoringMath.sol";
+import "./libraries/Boring/BoringRebase.sol";
 
 import "../interfaces/chainlink/AggregatorInterface.sol";
 import "../interfaces/swap/ISwap.sol";
+import "../interfaces/swap/ICurveFI.sol";
 import "../interfaces/yearn/IBaseFee.sol";
-import "../interfaces/yearn/IOSMedianizer.sol";
 import "../interfaces/yearn/IVault.sol";
 
+
+interface IBentoBoxV1 {
+    function balanceOf(IERC20, address) external view returns (uint256);
+
+    function transfer(
+        IERC20 token,
+        address from,
+        address to,
+        uint256 share
+    ) external;
+
+    function deposit(
+        IERC20 token,
+        address from,
+        address to,
+        uint256 amount,
+        uint256 share
+    ) external payable returns (uint256 amountOut, uint256 shareOut);
+
+    function setMasterContractApproval(
+        address user,
+        address masterContract,
+        bool approved,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external;
+
+    function toAmount(
+        IERC20 token,
+        uint256 share,
+        bool roundUp
+    ) external view returns (uint256 amount);
+
+    function toShare(
+        IERC20 token,
+        uint256 amount,
+        bool roundUp
+    ) external view returns (uint256 share);
+
+    function totals(IERC20) external view returns (Rebase memory totals_);
+
+    function withdraw(
+        IERC20 token_,
+        address from,
+        address to,
+        uint256 amount,
+        uint256 share
+    ) external returns (uint256 amountOut, uint256 shareOut);
+}
+
+interface IAbracadabra {
+    function COLLATERIZATION_RATE() external view returns (uint256);
+
+    function bentoBox() external view returns (IBentoBoxV1);
+
+    function masterContract() external view returns (address);
+
+    function addCollateral(
+        address to,
+        bool skim,
+        uint256 share
+    ) external;
+
+    function removeCollateral(address to, uint256 share) external;
+
+    function borrow(address to, uint256 amount)
+        external
+        returns (uint256 part, uint256 share);
+
+    function repay(
+        address to,
+        bool skim,
+        uint256 part
+    ) external returns (uint256 amount);
+
+    function exchangeRate() external view returns (uint256);
+
+    function userBorrowPart(address) external view returns (uint256);
+
+    function userCollateralShare(address) external view returns (uint256);
+
+    function totalBorrow() external view returns (Rebase memory totals);
+
+    function collateral() external view returns (address);
+
+    function accrue() external;
+}
+
+
```

Comment:
@newmickymousse let's place interfaces in their own directory so we clean the strategy contract a bit.

Reply:
I have to break the BoringRebase library to do this. That's why I added these two interfaces to the main contract.